pkg/autodiscovery/providers/kubelet.go

```diff
@@ -90,17 +97,23 @@ func parseKubeletPodlist(podlist []*kubelet.Pod) ([]integration.Config, error) {
 				legacyPodAnnotationPrefix, pod.Metadata.Name, newPodAnnotationPrefix)
 		}
 
+		adErrors := []error{}
+		containerIdentifiers := map[string]bool{}
+		containerNames := map[string]bool{}
```

Comment:
since we just need the `set` data structure here, let's use the type `map[string]struct{}` instead of `map[string]bool{}` (more details [here](https://stackoverflow.com/questions/22770114/any-difference-in-using-an-empty-interface-or-an-empty-struct-as-a-maps-value))